src/main/java/com/github/susom/database/Database.java

```diff
@@ -198,6 +198,41 @@
    */
   void dropTableQuietly(String tableName);
 
+  /**
+   * Convenience method to check if a table or view exists so that caller can decide
+   * whether to create or update a table.  The lookup is case insensitive using upper or
+   * lower case depending on the DB in use, unless the tableName is double quoted.
+   * The default catalog and schema will be used.
+   *
+   * @param tableName the table to be checked
+   * @return true if the table or view exists
+   */
+  boolean tableExists(@Nonnull String tableName);
+
+  /**
+   * Convenience method to check whether a table or view exists or not.
+   * This check is case-insensitive unless the tablename is double quoted.
+   * Lookup will be done using the default case for the DB in use.
+   * The default catalog will be used.
+   *
+   * @param tableName the table to be checked
+   * @param schemaName the schema expected to contain the table
+   * @return true if the table or view exists
+   */
+  boolean tableExists(@Nonnull String tableName, String schemaName);
+
+  /**
+   * Convenience method to check whether a table or view exists or not.
+   * This check is case-insensitive unless the table name is double quoted.
+   * Lookup will be done using the default case for the DB in use.
+   *
+   * @param tableName the table to be checked
+   * @param schemaName the schema expected to contain the table
+   * @param catalogName the catalog name associated with the schema and table
+   * @return true if the table or view exists
+   */
+  boolean tableExists(@Nonnull String tableName, String schemaName, String catalogName);
```

Comment:
Do we actually have a case where we need this? We don't really allow specifying catalog anywhere else, unless you drop down to specifying within your ddl.

Reply:
The only use case that I have is for tableExists(tableName).  I included tableExists(tableName, schemaName), and tableExists(tableName, schemaName, CatalogName) because I was trying to make it fit with your philosophy of general purpose use.  I am happy to drop that if it's more consistent with the API.